core/field_colour.js

```diff
@@ -148,6 +149,13 @@ Blockly.FieldColour.COLOURS = goog.ui.ColorPicker.SIMPLE_GRID_COLORS;
 Blockly.FieldColour.COLUMNS = 7;
 
 /**
+ * Function to be called if eyedropper can be activated.
+ * If defined, an eyedropper button will be added to the color picker.
+ * The button calls this function with a callback to update the field value.
+ */
+Blockly.FieldColour.activateEyedropper = null;
```

Comment:
Would this be global for all color picker fields, no matter what block they're attached to? I'm worried that we don't actually want that. It might be fine in the context of Scratch, but for scratch-blocks, it'd be nice if we could be more flexible. I think it is unfortunate that the colors themselves are also global across all color fields and I'd like to avoid adding more if we can.

Reply:
Recapping what we discussed offline: I think we need to take a scratch-3-first approach here, since we can't really figure out any situations that would require this. Once they arise we may be able to re-evaluate. 

One switch we may experiment with is a color field that goes directly to color picking mode, but we can add that later if we find it is useful.

Reply:
The problem with this approach is once you expose this "api" publicly, you can't really get rid of it because other developers may be using it and we have no way of knowing.

But redesigning the field API isn't happening quickly... This isn't great either but what do you think about you naming Blockly.FieldColour.activateEyedropper with a trailing underscore and marking it private with a note?  You'll then be violating private by using it in Scratch gui or vm, but it at least gives you the flexibility to rip it out later without pulling the rug out from other developers.